### PR TITLE
Adds an achievement potion awarding permenant cosmetic items.

### LINKED ISSUE #68762: 5000 hc random points does not give you a golden wheelchair with the parapalegic perk
Reporting client version: 514.1584

## Round ID:
[187308](https://scrubby.melonmesa.com/round/187308)

## Testmerges:
- [Refactors SM Delaminations to be strategies](https://www.github.com/tgstation/tgstation/pull/68599)

## Reproduction:
image

I did my time give me the goods

this is my first round after getting the points required and I think I set paraplegic in the set up screen as the round started. I latejoined instead of starting the shift in vivo

NEXT round I was shift start and still did not have the wheelchair

### PR DESCRIPTION
**Pull request by orthography**
## About The Pull Request

Adds a potion that converts certain items into their gold equivalents for having a certain achievement threshold.

- Bartenders with 5000+ bar tourists points will give them the bartender's midas potion, for the golden nanotransen shaker.

- Cooks with 5000+ kitchen tourists points will give them the cook's midas potion, for the golden knife.

- Earing 5000+ hardcore points will now give them the martyr's midas potion, for the golden wheelchair.

## Why It's Good For The Game

Gives purpose to the tourists served achievement, especially as during slow shifts (more so for bartenders) there is nothing to encourage players to use the tourist bots. This should make the gameplay loop a little more player sided giving an actual objective for service staff to achieve to flex their expertise and devotion to the job, ~and then have the item stolen in the first 5 minutes by Tider McToolbox~

Fixes #68762

## Changelog

:cl:
add: Nanotransen has begun a culinary awards program, high achieving service staff will receive a solid gold trophy to use during their shifts to thank them for their tireless work. Please do not break it.
imageadd: adds golden versions of the knife and shaker.
code: adds achievement checks in the bartender.dm and cook.dm
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
GAGSify those sprites 😈

**Comment by Pepsilawn** [comment]
Cooks don't normally start with knives as they must be bought from the vendor, I'm not sure that's the right item to give them

**Comment by orthography** [comment]
> Cooks don't normally start with knives as they must be bought from the vendor, I'm not sure that's the right item to give them

Since knifes are so incredibly cheap for chefs, and it's so necessary for their job, I personally don't see an issue with them spawning with it.

It'd be hell, but I *could* try making an item that converts the knife/shaker into the gold variant.

**Comment by wesoda25** [comment]
Encourages single player gameplay 👎

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -23,7 +23,10 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 	var/lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
 	///Icon file for right inhand overlays
 	var/righthand_file = 'icons/mob/inhands/items_righthand.dmi'
-
+	///Achievement items.
+	var/goldenk = FALSE
+	var/goldens = FALSE
+	var/goldenw = FALSE
```
[the remark this answers is not in the document]
Comment (orthography) [reply code/game/objects/items.dm]: https://github.com/tgstation/tgstation/pull/68758/commits/0cfc54cb5b2d1f66febf7e9690fc4067fea3b7a2

File: code/modules/jobs/job_types/_job.dm [hunk 2]
```diff
@@ -320,6 +320,11 @@
 
 	if(client?.is_veteran() && client?.prefs.read_preference(/datum/preference/toggle/playtime_reward_cloak))
 		neck = /obj/item/clothing/neck/cloak/skill_reward/playing
+/datum/outfit/job/pre_equip(mob/living/carbon/human/H, visualsOnly)
+	. = ..()
+	if(H.client?.get_award_status(HARDCORE_RANDOM_SCORE) >= 5000)
+		backpack_contents[/obj/item/achievement_potion/hardcore] = 1
```
[the remark this answers is not in the document]
Comment (orthography) [reply code/modules/jobs/job_types/_job.dm]: https://github.com/tgstation/tgstation/pull/68758/commits/6da0254f4d83863ada166602074ed60275e0df61

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/jobs/job_types/bartender.dm [hunk 3]
```diff
@@ -58,6 +58,11 @@
 	glasses = /obj/item/clothing/glasses/sunglasses/reagent
 	shoes = /obj/item/clothing/shoes/laceup
 
+/datum/outfit/job/bartender/pre_equip(mob/living/carbon/human/H, visualsOnly)
```
[the remark this answers is not in the document]
Comment (orthography) [reply code/modules/jobs/job_types/bartender.dm]: https://github.com/tgstation/tgstation/pull/68758/commits/9598d1ee1661ed7ed1c5196905b4d22ab6a51033

File: code/modules/jobs/job_types/_job.dm [hunk 4]
```diff
@@ -321,6 +321,11 @@
 	if(client?.is_veteran() && client?.prefs.read_preference(/datum/preference/toggle/playtime_reward_cloak))
 		neck = /obj/item/clothing/neck/cloak/skill_reward/playing
 
+/datum/outfit/job/pre_equip(mob/living/carbon/human/H, visualsOnly)
+	. = ..()
```
[the remark this answers is not in the document]
Comment (orthography) [reply code/modules/jobs/job_types/_job.dm]: https://github.com/tgstation/tgstation/pull/68758/commits/9598d1ee1661ed7ed1c5196905b4d22ab6a51033

File: code/game/objects/items/kitchen.dm [hunk 5]
```diff
@@ -79,6 +79,13 @@
 	name = "kitchen knife"
 	desc = "A general purpose Chef's Knife made by SpaceCook Incorporated. Guaranteed to stay sharp for years to come."
 
+/obj/item/knife/kitchen/attackby(obj/item/achievement_potion/I, mob/user)
+	if(istype(I) && I.golden_knife == TRUE)
+		user.visible_message(span_notice("[user] pours the potion onto [src]. It glows brightly before turning into gold!"))
+		new /obj/item/knife/gold(loc)
```
[the remark this answers is not in the document]
Comment (orthography) [reply code/game/objects/items/kitchen.dm]: https://github.com/tgstation/tgstation/pull/68758/commits/05bbeb29a246397adc9ec6b2142c1e1e7341b64a

File: code/game/objects/items/kitchen.dm [hunk 6]
```diff
@@ -79,6 +79,21 @@
...
+/obj/item/knife/kitchen/attackby(obj/item/attacking_item, mob/user, params)
+	. = ..()
+	if(.)
+		return TRUE
+
+	if(!istype(attacking_item, /obj/item/achievement_potion/cook))
+		return FALSE
+
+	user.visible_message(span_notice("[user] pours [attacking_item] onto [src]. It glows brightly before turning into gold!"))
+	var/obj/item/knife/golden = new(drop_location())
```
Comment (ShizCalev) [inline code/game/objects/items/kitchen.dm:91]: transformations should be loc

**Comment by NamelessFairy** [comment]
I really like the concept, I agree with your suggestion about making these into converters rather than spawning with the item, (latejoin cooks spawning armed isn't a cosmetic change). I'm not a fan of the goals for the bartender/cook though, it should be something more interactive, maybe mix 10 of every possible drink or just mix x number of drinks overall (I can see this being exploited though) By doing an objective like this your effectively engaging in the same thing that your job tasks people with but this time its not exclusive to feeding NPCs. Also would like to know how difficult the 3 of these objectives are, I don't play any of them but how long are these realistically going to take.

**Comment by orthography** [comment]
> I really like the concept, I agree with your suggestion about making these into converters rather than spawning with the item, (latejoin cooks spawning armed isn't a cosmetic change). I'm not a fan of the goals for the bartender/cook though, it should be something more interactive, maybe mix 10 of every possible drink or just mix x number of drinks overall (I can see this being exploited though) By doing an objective like this your effectively engaging in the same thing that your job tasks people with but this time its not exclusive to feeding NPCs. Also would like to know how difficult the 3 of these objectives are, I don't play any of them but how long are these realistically going to take.

Hi, I could understand your concern with the knife which is why it's attached to a potion now. As of now, there are about 4 /tg/ players who have unlocked the golden wheelchair doing hardcore random, it is definitely supposed to be a hard task, and because of that I don't feel comfortable changing the requirements without the original owners permission. (https://github.com/tgstation/tgstation/pull/50606)

The tourists were achievements that already existed keeping score in a (relatively) fair way, the more complex the drink, the more points you get. Furthermore, they're a dependable source of points. As mentioned above, especially for bartender, it's unlikely that you get much action at all during your shift. Unless you turn to bounties or NPCs, you might have the mime ask for silencer, and sec as for quad.

Hope that clarifies it a bit more.

**Review by Twaticus — APPROVED** [review]

**Comment by optimumtact** [comment]
@Mothblocks @ninjanomnom thoughts?

I'm mixed on this, especially as we already have skill cloaks.

**Comment by NamelessFairy** [comment]
> @Mothblocks @ninjanomnom thoughts?
> 
> I'm mixed on this, especially as we already have skill cloaks.

With the exception of the playtime skill cloak the skill cloaks are per round, I don't know the exact numbers but e.g. mine x number of rocks during a shift get a skill cloak for that shift, you do not get the skill cloak next shift and need to earn it again. These are larger Multi-round objectives that give you a permanent cosmetic item.

File: code/game/objects/items/kitchen.dm (hunk 6 above)
Comment (Mothblocks) [inline code/game/objects/items/kitchen.dm:91]: This doesn't look like you're using the golden subtype?

**Comment by Mothblocks** [comment]
@optimumtact Skill capes aren't really equivalent in the moment I don't think because they're temporary, and aside from hyper-gaming strats, you usually only obtain them when the round is already nearing its conclusion (obviously adjustable).

I think the way I would best prefer to see these is with challenging, but more reachable, requirements, and if it expires after a while, like a week or so.

Also a bit negative about attaching it to the bar bots, which although I'm generally a fan of them, I don't think I want to see them have a stronger presence since I don't want to give people a reason to prefer them over real human players. Maybe even just attaching them to the actual skills would be a good idea if we go with my expiration suggestion.

**Comment by orthography** [comment]
> @optimumtact Skill capes aren't really equivalent in the moment I don't think because they're temporary, and aside from hyper-gaming strats, you usually only obtain them when the round is already nearing its conclusion (obviously adjustable).
> 
> I think the way I would best prefer to see these is with challenging, but more reachable, requirements, and if it expires after a while, like a week or so.
> 
> Also a bit negative about attaching it to the bar bots, which although I'm generally a fan of them, I don't think I want to see them have a stronger presence since I don't want to give people a reason to prefer them over real human players. Maybe even just attaching them to the actual skills would be a good idea if we go with my expiration suggestion.

I totally understand your concerns with the barbots. My issue is that, especially for bartenders, there is almost no action in their shifts (interaction, antag potential, fights, etc). I was already looking into a way to make some of these lesser played jobs, (bartender, psych, curator) more active either by a harder injection into the round, or through giving them their own shit to do.

I could look into attaching the both the knife and shaker to more crew centric objective if you have one off the top of your head.

**Comment by orthography** [comment]
Whoops! Furthermore, especially unless they're not attached to bar bots, it seems unwise to make them expire. There was already some expressed concern in this PR that it could cause more single player gameplay than actual round engagement, which I could understand if it had to be renewed every shift.

Unless there are some divergent ideas, for the knife and shaker unlocks.

The possibilities, by mothblock points are.

Reward preservance.

- Keep it permanent
- Renew it every week
- Renew every month
- Renew every shift
- Must be top (x) on the leaderboard


New obtain methods.

- Start of shift bartender or chef must make some over the top drink/food and have at least one bite out of it. (Super burger, jungle juice, mega pie, narsie memosas, etc)
- Make x amount of drinks/food (by complexity.
- Serve x amount of crew mates
- Serve x amount of bar bots.

**Comment by LemonInTheDark** [comment]
sall good, you gonna reopen the pr?

**Pull request closed without merging**